messagebus/src/main/java/com/yahoo/messagebus/Sequencer.java

```diff
@@ -20,14 +20,21 @@ public class Sequencer implements MessageHandler, ReplyHandler {
     private final AtomicBoolean destroyed = new AtomicBoolean(false);
     private final MessageHandler sender;
     private final Map<Long, Queue<Message>> seqMap = new HashMap<>();
+    private final Messenger msn;
+    private final static Object BUSY = new Object();
+    private final static ThreadLocal<Object> isSending = new ThreadLocal<>();
```

Comment:
Would it suffice with `ThreadLocal<Boolean>` here without a separate sentinel object?

Reply:
Yes, it would. Fixed